### PR TITLE
Fix tagged vector assignment

### PR DESCRIPTION
**Pull request by EmilyBourne**

### CODE REVIEW

File: include/ddc/detail/tagged_vector.hpp [hunk 1]
```diff
@@ -293,6 +293,7 @@ class TaggedVector : public ConversionOperators<TaggedVector<ElementType, Tags..
             TaggedVector<ElementType, OTags...>&& other) noexcept
     {
         m_values = std::move(other.m_values);
```
Comment (tpadioleau) [inline include/ddc/detail/tagged_vector.hpp:295]:
```suggestion
```

File: tests/tagged_vector.cpp [hunk 2]
```diff
@@ -73,3 +73,21 @@ TEST(TaggedVector, Operators)
...
+    EXPECT_EQ(a.get<float>(), b.get<float>());
+}
+
+TEST(TaggedVector, ReorderingAssignment)
+{
+    ddc::detail::TaggedVector<int, double, float> a(1, 2);
+    ddc::detail::TaggedVector<int, float, double> c;
+    c = a;
+    EXPECT_EQ(a.get<double>(), c.get<double>());
+    EXPECT_EQ(a.get<float>(), c.get<float>());
```
Comment (tpadioleau) [inline tests/tagged_vector.cpp:92]:
```suggestion
    ddc::detail::TaggedVector<int, float, double> b;
    b = a;
    EXPECT_EQ(a.get<double>(), b.get<double>());
    EXPECT_EQ(a.get<float>(), b.get<float>());
```

**Review by tpadioleau — CHANGES_REQUESTED** [review]

File: include/ddc/detail/tagged_vector.hpp (hunk 1 above)
Comment (EmilyBourne) [reply include/ddc/detail/tagged_vector.hpp:295]: oops

File: tests/tagged_vector.cpp [hunk 3]
```diff
@@ -73,3 +73,39 @@ TEST(TaggedVector, Operators)
...
+    EXPECT_EQ(a.get<float>(), b.get<float>());
+}
+
+TEST(TaggedVector, MoveAssignment)
+{
+    ddc::detail::TaggedVector<int, double, float> a(1, 2);
+    ddc::detail::TaggedVector<int, double, float> b;
+    b = std::move(a);
+    EXPECT_EQ(a.get<double>(), b.get<double>());
+    EXPECT_EQ(a.get<float>(), b.get<float>());
```
Comment (tpadioleau) [inline tests/tagged_vector.cpp:101]: I agree that in the case of type `int` it shall work but we should not be using `a` after a move. I suggest:
```suggestion
    EXPECT_EQ(1, b.get<double>());
    EXPECT_EQ(2, b.get<float>());
```

File: tests/tagged_vector.cpp [hunk 4]
```diff
@@ -73,3 +73,39 @@ TEST(TaggedVector, Operators)
...
+}
+
+TEST(TaggedVector, ReorderingMoveAssignment)
+{
+    ddc::detail::TaggedVector<int, double, float> a(1, 2);
+    ddc::detail::TaggedVector<int, float, double> b;
+    b = std::move(a);
+    EXPECT_EQ(a.get<double>(), b.get<double>());
+    EXPECT_EQ(a.get<float>(), b.get<float>());
```
Comment (tpadioleau) [inline tests/tagged_vector.cpp:110]: Same
```suggestion
    EXPECT_EQ(1, b.get<double>());
    EXPECT_EQ(2, b.get<float>());
```

**Review by tpadioleau — CHANGES_REQUESTED** [review]

File: tests/tagged_vector.cpp (hunk 3 above)
Comment (EmilyBourne) [reply tests/tagged_vector.cpp:101]: Oops, I didn't really think. Just copy-pasted

Comment (EmilyBourne) [reply tests/tagged_vector.cpp]: Fixed [628025c](https://github.com/CExA-project/ddc/pull/210/commits/628025c3db4b224d4ce4584d299f19c4ef90c91c)

File: tests/tagged_vector.cpp (hunk 4 above)
Comment (EmilyBourne) [reply tests/tagged_vector.cpp]: Fixed [628025c](https://github.com/CExA-project/ddc/pull/210/commits/628025c3db4b224d4ce4584d299f19c4ef90c91c)

**Review by jbigot — APPROVED** [review]

**Review by tpadioleau — APPROVED** [review]
Looks good to me, thanks!

**Comment by jbigot** [comment]
This is a separate issue, but we should really align the constructors and assignment operators

**Pull request merged**